src/pytorch_metric_learning/wrappers/self_supervised_loss_wrapper.py

```diff
@@ -0,0 +1,114 @@
+            "TripletMarginLoss",
+            "NCALoss",
+            "TupletMarginLoss",
+        ]
+
+    @classmethod
+    def check_loss_support(cls, loss_name):
+        if loss_name not in cls.supported_losses():
+            raise Exception(f"SelfSupervisedLossWrapper not supported for {loss_name}")
+
+    def compute_loss(self, embeddings, ref_emb, *args):
+        """
+        embeddings: representations of the original set of inputs
+        ref_emb:    representations of an augmentation of the inputs.
+        *args:      variable length argument list, where each argument
+                    is an additional representation of an augmented version of the input.
+                    i.e. ref_emb2, ref_emb3, ...
+        """
+        embeddings_size = embeddings.size()
+        ref_emb_size = ref_emb.size()
+
+        embeddings_type = embeddings.type()
+        ref_emb_type = ref_emb.type()
+
+        embeddings_device = embeddings.get_device()
+        ref_emb_device = ref_emb.get_device()
+
+        if embeddings_size != ref_emb_size:
+            raise Exception(
+                f"Input parameters embeddings and ref_emb must be of the same size. Found '{embeddings_size}' and '{ref_emb_size}' instead."
+            )
+        if embeddings_type != ref_emb_type:
+            raise Exception(
+                f"Input parameters embeddings and ref_emb must be of the type. Found '{embeddings_type}' and '{ref_emb_type}' instead."
+            )
+        if embeddings_device != ref_emb_device:
+            raise Exception(
+                f"Input parameters embeddings and ref_emb must be on the same device. Found '{embeddings_device}' and '{ref_emb_device}' instead."
+            )
+
+        batch_size = embeddings_size[0]
+        labels = torch.arange(batch_size).type(embeddings_type).to(embeddings_device)
+        return self.loss(
+            embeddings=embeddings, labels=labels, ref_emb=ref_emb, ref_labels=labels
+        )
+
+    def forward(self, embeddings, ref_emb, *args):
+        return self.compute_loss(embeddings, ref_emb, *args)
+
+    @staticmethod
+    def get_blacklist():
+        return [
+            # Not loss functions
+            "BaseMetricLossFunction",
+            "EmbeddingRegularizerMixin",
+            "GenericPairLoss",
+            "MultipleLosses",
+            # Requires num_classes at initialization
+            "ArcFaceLoss",
+            "LargeMarginSoftmaxLoss",
+            "NormalizedSoftmaxLoss",
+            "ProxyAnchorLoss",
+            "ProxyNCALoss",
+            "SignalToNoiseRatioContrastiveLoss",
+            "SoftTripleLoss",
+            "SphereFaceLoss",
+            # Requires "num_pairs" at initialization
+            "NPairsLoss",
+            # Not applicable
+            "FastAPLoss",
+            "CentroidTripletLoss",
+            "VICRegLoss",
+            "AngularLoss",
+            "LiftedStructureLoss",
+            "NCALoss",
+            "TupletMarginLoss",
+        ]
```

Comment:
Still relevant?

Reply:
Nope, since we have a whitelist function now `supported_losses`